pipelinesScripts/validator/validator_test.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/jfrog/jfrog-cli-plugins-reg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+const testPluginRepo = "https://github.com/jfrog/jfrog-cli-plugins"
+
+func TestMain(m *testing.M) {
+	tempDir, err := setUp()
+	if err != nil {
+		log.Fatal(err)
+	}
+	m.Run()
+	if err := tearDown(tempDir); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func setUp() (string, error) {
+	return utils.CreatePLaygroundForJfrogCliTest()
+}
+
+func tearDown(tempDir string) error {
+	return os.RemoveAll(tempDir)
+}
+
+func TestValidateExtension(t *testing.T) {
+	descriptorName := "test_extention_plugin"
+	assert.NoError(t, execValidator(&utils.PluginDescriptor{}, descriptorName+".yml", validateExtension))
+	assert.Error(t, execValidator(&utils.PluginDescriptor{}, descriptorName, validateExtension))
+}
+
+func TestValidateDescriptorStructure(t *testing.T) {
+	pluginDescriptor := &utils.PluginDescriptor{
+		PluginName:  "My Plugin",
+		Version:     "v1.0.0",
+		Repository:  "www.myrepo.com",
+		Maintainers: []string{"First_maintainer", "Second_maintainer"},
+	}
+	descriptorName := "test_extention_plugin.yml"
+	assert.NoError(t, execValidator(pluginDescriptor, descriptorName, validateDescriptor))
+
+	// Validate mandatory fields
+	pluginDescriptorCopy := *pluginDescriptor
+	pluginDescriptorCopy.PluginName = ""
+	assert.Error(t, execValidator(&pluginDescriptorCopy, descriptorName, validateDescriptor))
+
+	pluginDescriptorCopy = *pluginDescriptor
+	pluginDescriptorCopy.Version = ""
+	assert.Error(t, execValidator(&pluginDescriptorCopy, descriptorName, validateDescriptor))
+
+	pluginDescriptorCopy = *pluginDescriptor
+	pluginDescriptorCopy.Repository = ""
+	assert.Error(t, execValidator(&pluginDescriptorCopy, descriptorName, validateDescriptor))
+
+	pluginDescriptorCopy = *pluginDescriptor
+	pluginDescriptorCopy.Maintainers = nil
+	assert.Error(t, execValidator(&pluginDescriptorCopy, descriptorName, validateDescriptor))
```

Comment:
Lets add also a test that expect error when `descriptor.Tag != ""` and `descriptor.Branch!= "" `

Reply:
I'd also prefer to check the error message and make sure it equals the expected